File: pkg/schemas/schemas.go
package schemas

import "time"

// User schemas
type User struct {
	ID         uint   `json:"id"`
	TelegramID int64  `json:"telegram_id"`
	FirstName  string `json:"first_name"`
	LastName   string `json:"last_name,omitempty"`
	Username   string `json:"username,omitempty"`
}

// Question schemas
type Question struct {
	ID      uint   `json:"id"`
	UserID  uint   `json:"user_id"`
	ClassID uint   `json:"class_id"`
	Text    string `json:"text"`
	User    User   `json:"user"`
}

// RSVP schemas
type RSVP struct {
	ID      uint   `json:"id"`
	UserID  uint   `json:"user_id"`
	ClassID uint   `json:"class_id"`
	Status  string `json:"status"`
	User    User   `json:"user"`
}

type RsvpRequest struct {
	TelegramID int64  `json:"telegram_id"`
	Status     string `json:"status"`
	FirstName  string `json:"first_name"`
	LastName   string `json:"last_name,omitempty"`
	Username   string `json:"username,omitempty"`
}

// Class schemas
type Class struct {
	ID          uint       `json:"id"`
	Topic       string     `json:"topic"`
	Description string     `json:"description"`
	ClassTime   time.Time  `json:"class_time"`
	CreatorID   uint       `json:"creator_id"`
	Creator     User       `json:"creator"`
	RSVPs       []RSVP     `json:"rsvps"`
	Questions   []Question `json:"questions"`
}

type ClassCreateRequest struct {
	Topic              string    `json:"topic"`
	Description        string    `json:"description"`
	ClassTime          time.Time `json:"class_time"`
	CreatorTelegramID  int64     `json:"creator_telegram_id"`
	CreatorFirstName   string    `json:"creator_first_name"`
	CreatorLastName    string    `json:"creator_last_name,omitempty"`
	CreatorUsername    string    `json:"creator_username,omitempty"`
}

type ClassUpdateRequest struct {
	UpdaterTelegramID int64       `json:"updater_telegram_id"`
	UpdateData        ClassUpdate `json:"update_data"`
}

type ClassUpdate struct {
	Topic       *string    `json:"topic,omitempty"`
	Description *string    `json:"description,omitempty"`
	ClassTime   *time.Time `json:"class_time,omitempty"`
}

File: app/schemas.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List

# Pydantic models for Users
class UserBase(BaseModel):
    telegram_id: int
    first_name: str
    last_name: Optional[str] = None
    username: Optional[str] = None

class UserCreate(UserBase):
    pass

class User(UserBase):
    id: int

    class Config:
        from_attributes = True

# Pydantic models for Questions
class QuestionBase(BaseModel):
    text: str

class QuestionCreate(QuestionBase):
    pass

class Question(QuestionBase):
    id: int
    user_id: int
    class_id: int
    user: User

    class Config:
        from_attributes = True

# Pydantic models for RSVPs
class RSVPBase(BaseModel):
    status: str

class RSVPCreate(RSVPBase):
    pass

class RsvpRequest(BaseModel):
    telegram_id: int
    status: str # "yes", "no", "tentative"
    first_name: str
    last_name: Optional[str] = None
    username: Optional[str] = None

class RSVP(RSVPBase):
    id: int
    user_id: int
    class_id: int
    user: User

    class Config:
        from_attributes = True

# Pydantic models for Classes
class ClassBase(BaseModel):
    topic: str
    description: str
    class_time: datetime

class ClassCreate(ClassBase):
    pass

class ClassCreateRequest(ClassBase):
    creator_telegram_id: int
    creator_first_name: str
    creator_last_name: Optional[str] = None
    creator_username: Optional[str] = None

class ClassUpdate(BaseModel):
    topic: Optional[str] = None
    description: Optional[str] = None
    class_time: Optional[datetime] = None

class ClassUpdateRequest(BaseModel):
    updater_telegram_id: int
    update_data: ClassUpdate

class Class(ClassBase):
    id: int
    creator_id: int
    creator: User
    rsvps: List[RSVP] = []
    questions: List[Question] = []

    class Config:
        from_attributes = True
